File: fixtures/scale/ruby/large/lib/check_pearsonhash.rb
require_relative 'pearsonhash_digest'
require_relative 'pearsonhash_digest_pair'
require_relative 'corpus_hash'

def check_pearsonhash
  corpus_hash.each do |text|
    first = pearsonhash_digest(text)
    second = pearsonhash_digest(text)
    return false if first != second || first.length != 8
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_pearsonhash.py
from cipherlab.ciphers.pearsonhash import pearsonhash_digest
from cipherlab.specs.pearsonhash_spec import pearsonhash_spec
from cipherlab.corpus.corpus_hash import corpus_hash


def check_pearsonhash():
    spec = pearsonhash_spec()
    for text in corpus_hash():
        first = pearsonhash_digest(text)
        second = pearsonhash_digest(text)
        if first != second or len(first) != 8:
            return False
    return spec["category"] == "hash"
